src/components/Calendar/index.tsx

```diff
@@ -135,14 +169,88 @@ export default function Calendar({
       });
   });
 
+  // Sort the events according to size of meeting.
+  const sortedEvents = events
+    .slice()
+    .sort(
+      (a: Immutable<Event>, b: Immutable<Event>) =>
+        a.period.start - b.period.start ?? 0
+    );
+
+  // Populates eventSizeInfo by iteratively finding the next time block's
+  // rowSize and rowIndex (1 more than greatest of already processed connected
+  // blocks), updating the processed connected blocks to match its rowSize (both events and crns)
+  sortedEvents.forEach((event) => {
```

Comment:
There's quite a lot of overlap between this populating algorithm for `eventSizeInfo` and `crnSizeInfo`. Can we consolidate it into a method?